src/custom/state/orders/updaters/CancelledOrdersUpdater.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+import { useActiveWeb3React } from 'hooks/web3'
+
+import { useCancelledOrders, useFulfillOrdersBatch } from 'state/orders/hooks'
+import { ApiOrderStatus } from 'state/orders/utils'
+import { OrderFulfillmentData } from 'state/orders/actions'
+import { OPERATOR_API_POLL_INTERVAL } from 'state/orders/consts'
+
+import { SupportedChainId as ChainId } from 'constants/chains'
+import { CANCELLED_ORDERS_PENDING_TIME } from 'constants/index'
+
+import { fetchOrderPopupData, OrderLogPopupMixData } from 'state/orders/updaters/utils'
+
+/**
+ * Updater for cancelled orders.
+ *
+ * Similar to Event updater, but instead of watching pending orders, it watches orders that have been cancelled
+ * in the last 5 min.
+ *
+ * Whenever an order that was cancelled but has since been fulfilled, trigger a state update
+ * and a popup notification, changing the status from cancelled to fulfilled.
+ *
+ * It's supposed to fix race conditions between the api accepting a cancellation while a solution was already
+ * submitted to the network by a solver.
+ * Due to the network's nature, we can't tell whether an order has been really cancelled, so we prefer to wait a short
+ * period and say it's cancelled even though in some cases it might actually be filled.
+ */
+export function CancelledOrdersUpdater(): null {
```

Comment:
Ideally, in one previous PR you moved these other PRs code so it doesn't look like is part of this one. 
I'm not reviewing this code cause im assuming theres no changes in the `Updaters` (only in unfillable)

Reply:
I know, I know https://github.com/gnosis/cowswap/pull/1201#issuecomment-896372327

Was lazy, will not happen again, I learned my lesson, etc, etc